### PR TITLE
JAMES-4044 Add a cache for LDAP attribute matchers

### PR DESCRIPTION
**Pull request by chibenwa**
(help welcome to address review comments if needed)

### CODE REVIEW

File: server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/CacheSettings.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        int separatorPosition = condition.indexOf('=');
+        String key = condition.substring(0, separatorPosition).toLowerCase(Locale.US);
+        String value = condition.substring(separatorPosition + 1);
+        return Pair.of(key, value);
+    }
+
+    private final Duration duration;
+    private final int size;
+
+    public CacheSettings(Duration duration, int size) {
```
Comment (mbaechler) [inline server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/CacheSettings.java:75]: aren't we using java 21 yet? Why don't we use Records for such things?

Comment (chibenwa) [reply server/mailet/ldap/src/main/java/org/apache/james/transport/matchers/CacheSettings.java:75]: Fair ^^

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Green. Can we squash this?

**Comment by chibenwa** [comment]
Feel free yes

**Pull request merged**